### ISSUE TITLE
"Port 3389 is not in listening state" or "Tunnel destination is not allowed by client configuration."

### ISSUE THREAD

**Issue by wangshiok** [cloudradar-monitoring/rport#898]
**Describe the bug**
getting either one of above 2 errors when create RDP tunnel, fix is simple, simply restarted the rport client in services.
It will work one day or couple days, then same problem.

In case you know it. Is the bug related to `rportd` (server) or `rport` (client)?  'rport'(client)

**Environment**
Describe the environment where the bug occurs:
* RPort version - 0.9.5
* OS details (Linux distribution, and version or Windows version) - windows server 2016 Standard
* Is rportd running inside a docker container? no
* Are you using a reverse proxy? If yes, which one? no

**Log file**
Before submitting a bug report, inspect the log files of client and server.
`/var/log/rport/rportd.log` for the server and `/var/log/rport/rport.log` or `C:\Program Files\rport\rport.log` for the client.
[Increase the log level to debug](https://github.com/cloudradar-monitoring/rport/blob/0.9.0/rport.example.conf#L169-L171) and try to reproduce the error. 
Include relevant lines into your report.

**Screenshots**

**Additional context**

**Comment by Kelbethor** [comment]
Maybe same issue?:  #Tunnel destination is not allowed by client configuration after ONE Tunnel Works #813

**Comment by Squall-The-Crow** [comment]
Same here, and not only RDP but VNC tunnel send that error too

**Comment by thorstenkramm** [comment]
@Squall-The-Crow Have you checked RDP is definitely enabled and listening?
Test with PowerShell
```PowerShell
Test-NetConnection 127.0.0.1 -Port 3389
```

**Comment by nnmer** [comment]
I had same issue after setting up VNC to win device. Then cannot setup a tunnel to linux. Only after rport server was restarted  a tunnel to linux established (restart rport on client only was not helping).

**Comment by mmurche** [comment]
I have the same Problem now on Windows RDP connection, Yesterday i can connect without Problem. 
Today same error MSG, nothing is changed about last 12 Hours.

**Comment by philipp-schmidt** [comment]
Same issue here now with SSH connection. It was fine before reboot and now we can't create a tunnel.

**Comment by philipp-schmidt** [comment]
SSH Port 22 is definitely up, we have a separate way of connecting and it works fine.

**Comment by philipp-schmidt** [comment]
I can also confirm that a reset of rport via "sudo systemctl restart rport" just fixed this issue.

Considering that we have devices which we can only access via SSH over Rport this is definitely a very critical issue and could potentially lead to us being locked out of one of our devices indefinitely.

**Comment by dugen** [comment]
I can as well confirm that this happens on the current build 0.9.5.
A rportd service restart is necessary to bring functionality back up again

.

**Comment by thorstenkramm** [comment]
@dugen Thanks for reporting.
Can you test the following scenarios?
1. Create a tunnel, delete the tunnel properly using the delete button. Create a new tunnel.
  Does this work flawlessly?
2. Create the tunnel, don't use it, wait 5 minutes until the tunnel has been auto-deleted due to inactivity. Create a new tunnel.
  Does this scenario provoke the error you reported?

**Comment by dugen** [comment]
> @dugen Thanks for reporting. Can you test the following scenarios?
> 
> 1. Create a tunnel, delete the tunnel properly using the delete button. Create a new tunnel.
>    Does this work flawlessly?
> 2. Create the tunnel, don't use it, wait 5 minutes until the tunnel has been auto-deleted due to inactivity. Create a new tunnel.
>    Does this scenario provoke the error you reported?

I tried your testapproach.
It does not lead to the error reported.

I can say that it appears to me after longer time of inactivity on the rportd service.
E.g. the next morning no tunnel can be created anymore, restarting the rportd services fixes this instantly.

**Comment by dugen** [comment]
.I will have a look at the code and try to reproduce it

**Comment by brolifen** [comment]
Can confirm just had the same issue on a new setup trying to setup a tunnel for the first time using Ubuntu 22.04 and Rport 0.9.5.

**Comment by nuterum** [comment]
Unfortunately i have not had time to test in depth on my side but i had the same problem on two windows in my production environnement. I confirm that the service is still usable even with this problem on the tunnel side. For what i had time to check on the windows it's seem that the rport service lost the permission to use the port 3389 (on my case). Fortunately the service still working for the other part meaning you can send script or commands and i use that to restart the service. After the restart the service find the port open and can use it but i have not find what have cause this issue. (encounter on windows 10 with Rport 0.9.5) if i encounter it again i will try to check if i can find more information.

**Comment by tnickels** [comment]
I am having the same issue. After a period of inactivity of the rportd no longer allows tunnels to be created. Restart the rportd and it works again. Ubuntu 20.04.5

**Comment by glennonline** [comment]
I also have the same issue, restarting rport fixes the issue.In my case it's just with trying to use SSH, not RDP.

**Comment by martin-korf** [comment]
Debian 11, RPort 0.9.5 have same issue (btw same fix)

**Comment by wallbanger123** [comment]
Same issue. Restart fixes it, but this isn't sustainable. Debian 11 on Goole Could. Any updates on a fix?

**Comment by fromage9747** [comment]
I experience the same issue.  Restarting the RPORT container/server/instance solves the issue, at least until it occurs again.

**Comment by bloesss** [comment]
Same here. Very often this happens to new Clients that connect the First Time

**Comment by chriswzng** [comment]
Same here. It was working for about a month, then suddenly it stopped working. Restarting the rport service works.

**Comment by thorleifjacobsen** [comment]
Same here, just randomly popping up. Restarting stuff works.

**Comment by sandropardini** [comment]
We have solved this issue installing latest client 0.9.10 and the RDP is running without fail.

**Comment by amastelek** [comment]
Seeing the same issue on a new install of 0.9.5 that require a restart of rport.

**Comment by thorstenkramm** [comment]
We are at the finish line of the next release. 
It's still not advised to upgrade your server to 0.9.10 because it's incompatible with the frontend and the new frontend is not fully completed yet. 

🙏 But if you want to upgrade just a handful of clients to 0.9.10 and report if the issue is solved, would be super helpful. ♥️

**Comment by thorstenkramm** [comment]
This has now been fixed in the latest version 0.9.12 which now has a stable frontend. (Fingers crossed)

It's recommended to [upgrade your server](https://kb.rport.io/digging-deeper/server-maintenance/updating-rport#update-the-rport-server) and [all your clients](https://kb.rport.io/digging-deeper/server-maintenance/updating-rport#update-the-rport-clients) in any order.

🙏 Much appreciated if you confirm your issue has been solved by the update.

**Comment by bloesss** [comment]
As I mentioned, the Frontend seems very buggy...

**Comment by bloesss** [comment]
I dont know.
I installed the Frontend you told me to install, where can I see the version?
I did a rport Server update which I thought always does the latest stable version.

The frontend says 0.9.12 when I see under Settings/info.

And the frontend is unusable for us, does only show 22 Client Groups for example.

**Comment by amastelek** [comment]
> This has now been fixed in the latest version 0.9.12 which now has a stable frontend. (Fingers crossed)
> 
> It's recommended to [upgrade your server](https://kb.rport.io/digging-deeper/server-maintenance/updating-rport#update-the-rport-server) and [all your clients](https://kb.rport.io/digging-deeper/server-maintenance/updating-rport#update-the-rport-clients) in any order.
> 
> 🙏 Much appreciated if you confirm your issue has been solved by the update.

Will update on 20/5 and provide feedback.

**Comment by thorstenkramm** [comment]
@bloesss Check `/var/lib/rport/docrot/version.txt` Should be `rport-frontend-0.9.12-1-build-1129`

Please don't mix topics. This thread is about broken tunnelling.  
Any feedback and more bug reports are very welcome. If you cannot see more than 22 client groups, please open a new issue.

**Comment by linus-dev** [comment]
Same issue, but restarting server/client does **not** work in my case. Always the error: "Port 5900 is not in listening state."
Using version 0.9.12. 

Any suggestions why?

**Comment by martin-korf** [comment]
If rebooting didn't help, it looks like a problem with VNC? 
What version of VNC is installed on the PC? 

I remember that Allow loopback connections has to be enabled in TightVNC.
![image](https://github.com/realvnc-labs/rport/assets/17021120/36aa3a23-b249-4f25-9e9e-180f871b6d6c)

**Comment by linus-dev** [comment]
> If rebooting didn't help, it looks like a problem with VNC for me What version of VNC is installed on the PC?
> 
> I remember that Allow loopback connections has to be enabled in TightVNC. ![image](https://user-images.githubusercontent.com/17021120/240051640-36aa3a23-b249-4f25-9e9e-180f871b6d6c.png)

Im using Ubuntu Desktop on my client, how can I achieve this then? 
I will come back with my VNC version.

**Comment by d3al** [comment]
> We are at the finish line of the next release. It's still not advised to upgrade your server to 0.9.10 because it's incompatible with the frontend and the new frontend is not fully completed yet.
> 
> pray But if you want to upgrade just a handful of clients to 0.9.10 and report if the issue is solved, would be super helpful. hearts

This issue is not resolved. I'm using 9.12 client and server on Ubuntu 20.04/22.04 and I get this error. Sometimes the client will already have an RDP or SSH active. Sometimes there are no tunnels. Restarting the rport client service usually fixes the issue for awhile. Sometimes if I wait and try again in 5+ minutes the tunnel will be created. Client is using xrdp and openssh.

**Comment by igbenic** [comment]
It happens randomly, and then I use add anyway and it works and the RDP works too
